File: MiniDOMFinanziaria/Finanziaria/Pratiche/InfoPratica/CInfoPraticaCursor.vb
Imports minidom
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Anagrafica



Partial Public Class Finanziaria

    ''' <summary>
    ''' Cursor sulla tabella delle informazioni aggiuntive sulle pratice
    ''' </summary>
    ''' <remarks></remarks>
    Public Class CInfoPraticaCursor
        Inherits DBObjectCursorBase(Of CInfoPratica)

        Private m_IDPratica As New CCursorField(Of Integer)("IDPratica")
        Private m_IDCommerciale As New CCursorField(Of Integer)("IDCommerciale")
        'Private m_IDConsulente As New CCursorField(Of Integer)("IDConsulente")
        Private m_IDDistributore As New CCursorField(Of Integer)("IDDistributore")
        Private m_Trasferita As New CCursorField(Of Boolean)("Trasferita")
        Private m_TrasferitoDaURL As New CCursorFieldObj(Of String)("TrasferitoDaURL")
        Private m_DataTrasferimento As New CCursorField(Of Date)("DataTrasferimento")
        Private m_TrasferitoA As New CCursorFieldObj(Of String)("TrasferitoA")
        Private m_IDTrasferitoDa As New CCursorField(Of Integer)("IDTrasferitoDa")
        Private m_IDPraticaTrasferita As New CCursorField(Of Integer)("IDPraticaTrasferita")
        Private m_DataAggiornamentoPT As New CCursorField(Of Date)("DataAggiornamentoPT")
        Private m_EsitoAggiornamentoPT As New CCursorField(Of Integer)("EsitoAggiornamentoPT")
        Private m_Costo As New CCursorField(Of Decimal)("Costo")
        Private m_IDPraticaDiRiferimento As New CCursorField(Of Integer)("IDPraticaDiRiferimento")
        Private m_IDCorrezione As New CCursorField(Of Integer)("IDCorrezione")
        Private m_MotivoSconto As New CCursorFieldObj(Of String)("MotivoSconto")
        Private m_MotivoScontoDettaglio As New CCursorFieldObj(Of String)("MotivoScontoDettaglio")
        Private m_IDScontoAutorizzatoDa As New CCursorField(Of Integer)("IDScontoAutorizzatoDa")
        Private m_ScontoAutorizzatoIl As New CCursorField(Of Date)("ScontoAutorizzatoIl")
        Private m_ScontoAutorizzatoNote As New CCursorFieldObj(Of String)("ScontoAutorizzatoNote")
        Private m_IDCorrettaDa As New CCursorField(Of Integer)("IDCorrettaDa")
        Private m_DataCorrezione As New CCursorField(Of Date)("DataCorrezione")
        Private m_NoteAmministrative As New CCursorFieldObj(Of String)("NoteAmministrative")

        Private m_ValoreUpFront As New CCursorField(Of Decimal)("ValoreUpFront")
        Private m_ValoreProvvTAN As New CCursorField(Of Decimal)("ValoreProvvTAN")
        Private m_ValoreProvvAGG As New CCursorField(Of Decimal)("ValoreProvvAGG")
        Private m_ValoreProvvTOT As New CCursorField(Of Decimal)("ValoreProvvTOT")
        Private m_Flags As New CCursorField(Of Integer)("Flags")

        Public Sub New()
        End Sub

        Public ReadOnly Property ValoreUpFront As CCursorField(Of Decimal)
            Get
                Return Me.m_ValoreUpFront
            End Get
        End Property

        Public ReadOnly Property ValoreProvvTAN As CCursorField(Of Decimal)
            Get
                Return Me.m_ValoreProvvTAN
            End Get
        End Property

        Public ReadOnly Property ValoreProvvAGG As CCursorField(Of Decimal)
            Get
                Return Me.m_ValoreProvvAGG
            End Get
        End Property
        Public ReadOnly Property ValoreProvvTOT As CCursorField(Of Decimal)
            Get
                Return Me.m_ValoreProvvTOT
            End Get
        End Property
        Public ReadOnly Property Flags As CCursorField(Of Integer)
            Get
                Return Me.m_Flags
            End Get
        End Property


        Public ReadOnly Property IDCorrettaDa As CCursorField(Of Integer)
            Get
                Return Me.m_IDCorrettaDa
            End Get
        End Property

        Public ReadOnly Property DataCorrezione As CCursorField(Of Date)
            Get
                Return Me.m_DataCorrezione
            End Get
        End Property

        Public ReadOnly Property IDPratica As CCursorField(Of Integer)
            Get
                Return Me.m_IDPratica
            End Get
        End Property

        Public ReadOnly Property IDCommerciale As CCursorField(Of Integer)
            Get
                Return Me.m_IDCommerciale
            End Get
        End Property


        'Public ReadOnly Property IDConsulente As CCursorField(Of Integer)
        '    Get
        '        Return Me.m_IDConsulente
        '    End Get
        'End Property

        Public ReadOnly Property Trasferita As CCursorField(Of Boolean)
            Get
                Return Me.m_Trasferita
            End Get
        End Property

        Public ReadOnly Property TrasferitoDaURL As CCursorFieldObj(Of String)
            Get
                Return Me.m_TrasferitoDaURL
            End Get
        End Property

        Public ReadOnly Property DataTrasferimento As CCursorField(Of Date)
            Get
                Return Me.m_DataTrasferimento
            End Get
        End Property

        Public ReadOnly Property TrasferitoA As CCursorFieldObj(Of String)
            Get
                Return Me.m_TrasferitoA
            End Get
        End Property

        Public ReadOnly Property IDTrasferitoDa As CCursorField(Of Integer)
            Get
                Return Me.m_IDTrasferitoDa
            End Get
        End Property

        Public ReadOnly Property IDPraticaTrasferita As CCursorField(Of Integer)
            Get
                Return Me.m_IDPraticaTrasferita
            End Get
        End Property

        Public ReadOnly Property DataAggiornamentoPratica As CCursorField(Of Date)
            Get
                Return Me.m_DataAggiornamentoPT
            End Get
        End Property

        Public ReadOnly Property EsitoAggiornamentoPratica As CCursorField(Of Integer)
            Get
                Return Me.m_EsitoAggiornamentoPT
            End Get
        End Property

        Public ReadOnly Property Costo As CCursorField(Of Decimal)
            Get
                Return Me.m_Costo
            End Get
        End Property

        Public ReadOnly Property IDPraticaDiRiferimento As CCursorField(Of Integer)
            Get
                Return Me.m_IDPraticaDiRiferimento
            End Get
        End Property

        Public ReadOnly Property IDDistributore As CCursorField(Of Integer)
            Get
                Return Me.m_IDDistributore
            End Get
        End Property

        Public ReadOnly Property IDCorrezione As CCursorField(Of Integer)
            Get
                Return Me.m_IDCorrezione
            End Get
        End Property


        Public ReadOnly Property MotivoSconto As CCursorFieldObj(Of String)
            Get
                Return Me.m_MotivoSconto
            End Get
        End Property

        Public ReadOnly Property MotivoScontoDettaglio As CCursorFieldObj(Of String)
            Get
                Return Me.m_MotivoScontoDettaglio
            End Get
        End Property

        Public ReadOnly Property IDScontoAutorizzatoDa As CCursorField(Of Integer)
            Get
                Return Me.m_IDScontoAutorizzatoDa
            End Get
        End Property

        Public ReadOnly Property ScontoAutorizzatoIl As CCursorField(Of Date)
            Get
                Return Me.m_ScontoAutorizzatoIl
            End Get
        End Property

        Public ReadOnly Property ScontoAutorizzatoNote As CCursorFieldObj(Of String)
            Get
                Return Me.m_ScontoAutorizzatoNote
            End Get
        End Property

        Protected Overrides Function GetModule() As CModule
            Return Nothing
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_PraticheInfo"
        End Function

        Public Overrides Function InstantiateNew(dbRis As DBReader) As Object
            Return New CInfoPratica
        End Function

        Protected Overrides Function GetConnection() As CDBConnection
            Return Finanziaria.Database
        End Function

        Public ReadOnly Property NoteAmministrative As CCursorFieldObj(Of String)
            Get
                Return Me.m_NoteAmministrative
            End Get
        End Property

    End Class

End Class

File: MiniDOMFinanziaria/Finanziaria/Pratiche/InfoPratica/CInfoPraticaCursor.cs
using System;

namespace minidom
{
    public partial class Finanziaria
    {

        /// <summary>
    /// Cursor sulla tabella delle informazioni aggiuntive sulle pratice
    /// </summary>
    /// <remarks></remarks>
        public class CInfoPraticaCursor : Databases.DBObjectCursorBase<CInfoPratica>
        {
            private DBCursorField<int> m_IDPratica = new DBCursorField<int>("IDPratica");
            private DBCursorField<int> m_IDCommerciale = new DBCursorField<int>("IDCommerciale");
            // Private m_IDConsulente As New DBCursorField(Of Integer)("IDConsulente")
            private DBCursorField<int> m_IDDistributore = new DBCursorField<int>("IDDistributore");
            private DBCursorField<bool> m_Trasferita = new DBCursorField<bool>("Trasferita");
            private DBCursorStringField m_TrasferitoDaURL = new DBCursorStringField("TrasferitoDaURL");
            private DBCursorField<DateTime> m_DataTrasferimento = new DBCursorField<DateTime>("DataTrasferimento");
            private DBCursorStringField m_TrasferitoA = new DBCursorStringField("TrasferitoA");
            private DBCursorField<int> m_IDTrasferitoDa = new DBCursorField<int>("IDTrasferitoDa");
            private DBCursorField<int> m_IDPraticaTrasferita = new DBCursorField<int>("IDPraticaTrasferita");
            private DBCursorField<DateTime> m_DataAggiornamentoPT = new DBCursorField<DateTime>("DataAggiornamentoPT");
            private DBCursorField<int> m_EsitoAggiornamentoPT = new DBCursorField<int>("EsitoAggiornamentoPT");
            private DBCursorField<decimal> m_Costo = new DBCursorField<decimal>("Costo");
            private DBCursorField<int> m_IDPraticaDiRiferimento = new DBCursorField<int>("IDPraticaDiRiferimento");
            private DBCursorField<int> m_IDCorrezione = new DBCursorField<int>("IDCorrezione");
            private DBCursorStringField m_MotivoSconto = new DBCursorStringField("MotivoSconto");
            private DBCursorStringField m_MotivoScontoDettaglio = new DBCursorStringField("MotivoScontoDettaglio");
            private DBCursorField<int> m_IDScontoAutorizzatoDa = new DBCursorField<int>("IDScontoAutorizzatoDa");
            private DBCursorField<DateTime> m_ScontoAutorizzatoIl = new DBCursorField<DateTime>("ScontoAutorizzatoIl");
            private DBCursorStringField m_ScontoAutorizzatoNote = new DBCursorStringField("ScontoAutorizzatoNote");
            private DBCursorField<int> m_IDCorrettaDa = new DBCursorField<int>("IDCorrettaDa");
            private DBCursorField<DateTime> m_DataCorrezione = new DBCursorField<DateTime>("DataCorrezione");
            private DBCursorStringField m_NoteAmministrative = new DBCursorStringField("NoteAmministrative");
            private DBCursorField<decimal> m_ValoreUpFront = new DBCursorField<decimal>("ValoreUpFront");
            private DBCursorField<decimal> m_ValoreProvvTAN = new DBCursorField<decimal>("ValoreProvvTAN");
            private DBCursorField<decimal> m_ValoreProvvAGG = new DBCursorField<decimal>("ValoreProvvAGG");
            private DBCursorField<decimal> m_ValoreProvvTOT = new DBCursorField<decimal>("ValoreProvvTOT");
            private DBCursorField<int> m_Flags = new DBCursorField<int>("Flags");

            public CInfoPraticaCursor()
            {
            }

            public DBCursorField<decimal> ValoreUpFront
            {
                get
                {
                    return m_ValoreUpFront;
                }
            }

            public DBCursorField<decimal> ValoreProvvTAN
            {
                get
                {
                    return m_ValoreProvvTAN;
                }
            }

            public DBCursorField<decimal> ValoreProvvAGG
            {
                get
                {
                    return m_ValoreProvvAGG;
                }
            }

            public DBCursorField<decimal> ValoreProvvTOT
            {
                get
                {
                    return m_ValoreProvvTOT;
                }
            }

            public DBCursorField<int> Flags
            {
                get
                {
                    return m_Flags;
                }
            }

            public DBCursorField<int> IDCorrettaDa
            {
                get
                {
                    return m_IDCorrettaDa;
                }
            }

            public DBCursorField<DateTime> DataCorrezione
            {
                get
                {
                    return m_DataCorrezione;
                }
            }

            public DBCursorField<int> IDPratica
            {
                get
                {
                    return m_IDPratica;
                }
            }

            public DBCursorField<int> IDCommerciale
            {
                get
                {
                    return m_IDCommerciale;
                }
            }


            // Public ReadOnly Property IDConsulente As DBCursorField(Of Integer)
            // Get
            // Return Me.m_IDConsulente
            // End Get
            // End Property

            public DBCursorField<bool> Trasferita
            {
                get
                {
                    return m_Trasferita;
                }
            }

            public DBCursorStringField TrasferitoDaURL
            {
                get
                {
                    return m_TrasferitoDaURL;
                }
            }

            public DBCursorField<DateTime> DataTrasferimento
            {
                get
                {
                    return m_DataTrasferimento;
                }
            }

            public DBCursorStringField TrasferitoA
            {
                get
                {
                    return m_TrasferitoA;
                }
            }

            public DBCursorField<int> IDTrasferitoDa
            {
                get
                {
                    return m_IDTrasferitoDa;
                }
            }

            public DBCursorField<int> IDPraticaTrasferita
            {
                get
                {
                    return m_IDPraticaTrasferita;
                }
            }

            public DBCursorField<DateTime> DataAggiornamentoPratica
            {
                get
                {
                    return m_DataAggiornamentoPT;
                }
            }

            public DBCursorField<int> EsitoAggiornamentoPratica
            {
                get
                {
                    return m_EsitoAggiornamentoPT;
                }
            }

            public DBCursorField<decimal> Costo
            {
                get
                {
                    return m_Costo;
                }
            }

            public DBCursorField<int> IDPraticaDiRiferimento
            {
                get
                {
                    return m_IDPraticaDiRiferimento;
                }
            }

            public DBCursorField<int> IDDistributore
            {
                get
                {
                    return m_IDDistributore;
                }
            }

            public DBCursorField<int> IDCorrezione
            {
                get
                {
                    return m_IDCorrezione;
                }
            }

            public DBCursorStringField MotivoSconto
            {
                get
                {
                    return m_MotivoSconto;
                }
            }

            public DBCursorStringField MotivoScontoDettaglio
            {
                get
                {
                    return m_MotivoScontoDettaglio;
                }
            }

            public DBCursorField<int> IDScontoAutorizzatoDa
            {
                get
                {
                    return m_IDScontoAutorizzatoDa;
                }
            }

            public DBCursorField<DateTime> ScontoAutorizzatoIl
            {
                get
                {
                    return m_ScontoAutorizzatoIl;
                }
            }

            public DBCursorStringField ScontoAutorizzatoNote
            {
                get
                {
                    return m_ScontoAutorizzatoNote;
                }
            }

            protected override Sistema.CModule GetModule()
            {
                return null;
            }

            public override string GetTableName()
            {
                return "tbl_PraticheInfo";
            }

            public override object InstantiateNew(DBReader dbRis)
            {
                return new CInfoPratica();
            }

            protected override Databases.CDBConnection GetConnection()
            {
                return Database;
            }

            public DBCursorStringField NoteAmministrative
            {
                get
                {
                    return m_NoteAmministrative;
                }
            }
        }
    }
}